### PR TITLE
Click on empty taskbar space v1.1

### LINKED ISSUE #463: [Click on empty taskbar space] Adding taskbar item combining as an addition action option
@m1lhaus

I remember there being news articles about Microsoft supposedly having an ability where if you click on empty space on the taskbar it would toggle Never Combine <-> Always Combine option. This never came to be, or was ever confirmed. I believe this was only assumed by new authors when Microsoft was saying they would have some "smarter" implementation with "just one click".

I'm mostly referencing [Microsoft brings back Taskbar app ungrouping on Windows 11, but with a smart new twist](https://www.windowscentral.com/software-apps/windows-11/microsoft-brings-back-taskbar-app-ungrouping-on-windows-11-but-with-a-smart-new-twist)

Unsure if that would be something that could (or should for that matter) be added to the `Click on empty taskbar space` mod. Probably two options: one for toggling Always <-> When Full, and another for Always <-> Never.

### PR DESCRIPTION
**Pull request by m1lhaus**
Updated version of Click on empty taskbar space mod

Changelog:
- code reformatted by clang formatter - Visual Studio style (it makes the diff a pain in the ass I know and I am sorry for that - I will stick to this format from now on)
- removed 7+TT code used for Ctrl+Alt+Tab and Win+Tab (replaced by virtual keypress)
- implemented Open Start menu feature (sends Win keypress)
- implemented "Combine taskbar butons toggle" (works only on Windows 10 and newer Windows 11 versions) - closes #463
- prevent multiple double-clicks when tripple-clicking

Other features from https://github.com/m1lhaus/windhawk-mods/issues will be implemented later on.

### CODE REVIEW

File: mods/taskbar-empty-space-clicks.wh.cpp [hunk 1]
```diff
@@ -1356,111 +1319,255 @@ bool FindDesktopWindow() {
...
+{
+    HKEY hKey = NULL;
+    DWORD dwValue = 0;
+    DWORD dwBufferSize = sizeof(DWORD);
+    if (RegOpenKeyEx(HKEY_CURRENT_USER, TEXT("Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Advanced"),
+                     0, KEY_QUERY_VALUE, &hKey) == ERROR_SUCCESS)
+    {
+        if (RegQueryValueEx(hKey, TEXT("TaskbarGlomLevel"), NULL, NULL, (LPBYTE)&dwValue, &dwBufferSize) == ERROR_SUCCESS)
+        {
+            RegCloseKey(hKey);
```
Comment (m417z) [inline mods/taskbar-empty-space-clicks.wh.cpp]: You'd want to `RegCloseKey` even if `RegQueryValueEx` fails.

File: mods/taskbar-empty-space-clicks.wh.cpp [hunk 2]
```diff
@@ -1356,111 +1319,255 @@ bool FindDesktopWindow() {
...
+    if (option <= 2)
+    {
+        HKEY hKey = NULL;
+        if (RegOpenKeyEx(HKEY_CURRENT_USER, TEXT("Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Advanced"),
+                         0, KEY_SET_VALUE, &hKey) == ERROR_SUCCESS)
+        {
+            DWORD dwValue = option;
+            if (RegSetValueEx(hKey, TEXT("TaskbarGlomLevel"), 0, REG_DWORD, (BYTE *)&dwValue, sizeof(dwValue)) == ERROR_SUCCESS)
+            {
+                RegCloseKey(hKey);
```
Comment (m417z) [inline mods/taskbar-empty-space-clicks.wh.cpp]: You'd want to `RegCloseKey` even if `RegSetValueEx` fails.

File: mods/taskbar-empty-space-clicks.wh.cpp [hunk 3]
```diff
@@ -1356,111 +1319,255 @@ bool FindDesktopWindow() {
...
+    if (SendInput(NUM_KEYS * 2, input.get(), sizeof(input[0])) != (NUM_KEYS * 2))
+    {
+        Wh_Log(L"ERROR: Failed to send key input");
     }
 }
 
-void OpenTaskManager(HWND taskbarhWnd) {
+void SendCtrlAltTabKeypress()
+{
+    Wh_Log(L"Sending Ctrl+Alt+Tab kepress");
```
Comment (m417z) [inline mods/taskbar-empty-space-clicks.wh.cpp]: Typo (kepress).

File: mods/taskbar-empty-space-clicks.wh.cpp [hunk 4]
```diff
@@ -57,29 +65,46 @@ In case you are using old Windows taskbar on Windows 11 (**ExplorerPatcher** or
   $options:
   - ACTION_NOTHING: Nothing (default)
   - ACTION_SHOW_DESKTOP: Show desktop
-  - ACTION_ALT_TAB: Ctrl+Alt+Tab
+  - ACTION_CTRL_ALT_TAB: Ctrl+Alt+Tab
```
Comment (m417z) [inline mods/taskbar-empty-space-clicks.wh.cpp]: That will reset the configuration for uses who chose `ACTION_ALT_TAB`, as now it has a new identifier. Not necessarily too bad, just FYI, up to you whether you prefer a better name or not breaking the settings for users who selected this option.

Comment (m1lhaus) [reply mods/taskbar-empty-space-clicks.wh.cpp]: You are right. It is just a minor thing so I partially reverted the change.

**Pull request merged**